h2/connection.py

```diff
@@ -586,8 +593,11 @@ def initiate_upgrade_connection(self, settings_header=None):
             For servers, returns nothing.
         :rtype: ``bytes`` or ``None``
         """
-        frame_data = None
+        self.config.logger.debug(
+            "Upgrade connection.\nSettings frame: %s", self.local_settings
```

Comment:
I think structlog could work out rather nicely for the stream state machine, perhaps not for this PR but a future one.  @sigmavirus24 what is a scenario where a user would not want to use this?  Seems that if the messages were structured appropriately, it should be as good or better than the log messages as they are currently.